public/javascripts/controllers/buyer-ctrl.js

```diff
@@ -227,6 +230,89 @@ angular.module('jasmic.controllers')
                 $scope.demand.de_until = moment().add(7, 'days').toDate();
             };
 
+            $scope.editDemandForm = function (obj) {
+
```

Comment:
I noticed that most of your code has extra line breaks.  Is that your editor misbehaving?